qutip/nonmarkov/bofin.py

```diff
@@ -0,0 +1,1553 @@
+                    self.H_sys_list[0].shape[0] * self.H_sys_list[0].shape[0]
+                )
+            else:
+                self._sup_dim = H.shape[0] * H.shape[0]
+        else:
+            if self.isTimeDep:
+                self._sup_dim = (
+                    self.H_sys_list[0].shape[0]
+                )
+            else:
+                self._sup_dim = H.shape[0]
+
+    def steady_state(
+        self, max_iter_refine=100, use_mkl=False, weighted_matching=False
+    ):
+        """
+        Computes steady state dynamics
+        parameters:
+        max_iter_refine : Int
+            Parameter for the mkl LU solver. If pardiso errors are returned
+            this should be increased.
+        use_mkl : Boolean
+            Optional override default use of mkl if mkl is installed.
+        weighted_matching : Boolean
+            Setting this true may increase run time, but reduce stability
+            (pardisio may not converge).
+
+        Returns
+        -------
+        steady state :  Qobj
+            The steady state density matrix of the system
+        solution    :   Numpy array
+            Array of the the steady-state and all ADOs.
+            Further processing of this can be done with functions provided in
+            example notebooks.
+        """
+        nstates = self.nhe
+        sup_dim = self._sup_dim
+        n = int(np.sqrt(sup_dim))
+        L = deepcopy(self.RHSmat)
+
+        b_mat = np.zeros(sup_dim * nstates, dtype=complex)
+        b_mat[0] = 1.0
+
+        L = L.tolil()
+        L[0, 0: n ** 2 * nstates] = 0.0
+        L = L.tocsr()
+
+        # TODO: Replace with the standard QuTiP configuration options for MKL.
+        if settings.has_mkl & use_mkl:
+            # TODO: Replace with logging?
+            print("Using Intel mkl solver")
+            from qutip._mkl.spsolve import mkl_spsolve
+
+            L = L.tocsr() + sp.csr_matrix((
+                np.ones(n),
+                (np.zeros(n), [num * (n + 1) for num in range(n)])
+            ), shape=(n ** 2 * nstates, n ** 2 * nstates))
+
+            L.sort_indices()
+
+            solution = mkl_spsolve(
+                L,
+                b_mat,
+                perm=None,
+                verbose=True,
+                max_iter_refine=max_iter_refine,
+                scaling_vectors=True,
+                weighted_matching=weighted_matching,
+            )
+
+        else:
+
+            L = L.tocsc() + sp.csc_matrix((
+                np.ones(n),
+                (np.zeros(n), [num * (n + 1) for num in range(n)])
+            ), shape=(n ** 2 * nstates, n ** 2 * nstates))
+
+            # Use superLU solver
+
+            LU = splu(L)
+            solution = LU.solve(b_mat)
+
+        dims = self.H_sys.dims
+        data = dense2D_to_fastcsr_fmode(vec2mat(solution[:sup_dim]), n, n)
+        data = 0.5 * (data + data.H)
+
+        solution = solution.reshape((nstates, self.H_sys.shape[0] ** 2))
```

Comment:
if H is a superoperator, this fails.....  safer to use 

`solution = solution.reshape((nstates, sup_dim))`

Reply:
Thanks. Fixed.